script/subtitles/imscsubtitles.js

```diff
@@ -84,24 +146,43 @@ define('bigscreenplayer/subtitles/imscsubtitles',
       }
 
       function update (currentTime) {
-        var subtitlesIndex = nextSubtitleIndex(currentTime);
-        var generateAndRender = subtitlesIndex !== previousSubtitlesIndex;
+        var segment = getSegmentToRender(currentTime);
 
-        if (generateAndRender) {
-          render(currentTime);
-          previousSubtitlesIndex = subtitlesIndex;
+        if (segment) {
+          DebugTool.keyValue({key: 'segmentToRender', value: 'true'});
+          render(currentTime, segment.xml);
+        } else {
+          DebugTool.keyValue({key: 'segmentToRender', value: 'false'});
         }
       }
 
-      function render (currentTime) {
+      function getSegmentToRender (currentTime) {
+        var segment;
+
+        for (var i = 0; i < segments.length; i++) {
+          for (var j = 0; j < segments[i].times.length; j++) {
+            var lastOne = segments[i].times.length === j + 1;
+            if (currentTime >= segments[i].times[j] && (lastOne || currentTime < segments[i].times[j + 1]) && segments[i].previousSubtitleIndex !== j && segments[i].times[j] !== 0) {
+              segment = segments[i];
+              currentSegmentRendered = segments[i];
+              segments[i].previousSubtitleIndex = j;
+              break;
+            }
+          }
+        }
+
+        return segment;
+      }
+
+      function render (currentTime, xml) {
         removeCurrentSubtitlesElement();
 
         currentSubtitlesElement = document.createElement('div');
         currentSubtitlesElement.id = 'bsp_subtitles';
         currentSubtitlesElement.style.position = 'absolute';
         parentElement.appendChild(currentSubtitlesElement);
 
-        renderHTML(xml, currentTime, currentSubtitlesElement, imscRenderOpts, parentElement.clientHeight, parentElement.clientWidth);
+        return renderHTML(xml, currentTime, currentSubtitlesElement, imscRenderOpts, parentElement.clientHeight, parentElement.clientWidth);
```

Comment:
Any particular reason to return the result of the function here?

Reply:
No, good spot. A hangover from when we did need to with the way we were checking if something had already been rendered.